aten/src/ATen/core/boxing/impl/boxing.h

```diff
@@ -223,7 +263,7 @@ struct BoxedKernelWrapper<
     torch::jit::push(stack, std::forward<OtherArgs>(otherArgs)...);
 
     (*boxed_kernel_func)(functor, opHandle, &stack);
-    TORCH_INTERNAL_ASSERT_DEBUG_ONLY(
+    TORCH_CHECK(
```

Comment:
Did you measure if the removal of DEBUG_ONLY has a perf impact?

Reply:
I didn't, and I expect it does have some marginal impact, but (a) I looked around and kernels seem to use `TORCH_CHECK` pretty liberally for similar purposes (e.g. argument checking), so it seems idiomatic... and (b) the alternative seems like bad user experience (stack underflows or silent throwing away values).

Also maybe worth noting that this codepath won't get exercised by boxed fallbacks or profiling, the two use cases I can think of where the speed probably matters most.